sympy/polys/polytools.py

```diff
@@ -6218,8 +6216,45 @@ def sqf_list(f, *gens, **args):
     (2, [(x + 1, 2), (x + 2, 3)])
 
     """
-    return _generic_factor_list(f, gens, args, method='sqf')
-
+    options.allowed_flags(args, ['polys'])
+    opt = options.build_options(gens, args)
+    e = together(f)
+    gen_set = set()
+    coeff = 1
+    facs = []
+    for fac in Mul.make_args(e):
+        b, e = (fac.base, fac.exp) if fac.is_Pow else (fac, S.One)
+        if e < 0:
+            raise PolynomialError("a univariate polynomial expected")
+        try:
+            poly, _ = _poly_from_expr(b, opt)
+        except PolificationFailed:
+            coeff *= b
+        else:
+            gen_set.add(poly.gens)
```

Comment:
`poly.gens` is a tuple. We should add the elements of the tuple.

Reply:
oh yes, will correct